debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -163,6 +168,14 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
         finally {
             if (replicationConnection != null) {
                 LOGGER.debug("stopping streaming...");
+                // stop the keep alive thread, this also shuts down the
+                // executor pool
+                replicationStream.updateAndGet((stream) -> {
```

Comment:
The passed function should be side-effect free. In the original implementation it was done like that:

```
ReplicationStream stream = this.replicationStream.get();
// if we have a stream, ensure that it has been stopped
if (stream != null) {
    stream.stopKeepAlive();
}
```

Does speak anything against using this again?